runtime/oti/VMHelpers.hpp

```diff
@@ -2048,13 +2049,65 @@ class VM_VMHelpers
 #endif /* JAVA_SPEC_VERSION > 11 */
 	}
 
+#if JAVA_SPEC_VERSION >= 19
+	static VMINLINE bool
+	tryEnterConcurrentContinuationMonitor(J9VMContinuation *continuation)
+	{
+		volatile uintptr_t *localAddr = (volatile uintptr_t *) &continuation->carrierThread;
+		uintptr_t oldValue = *localAddr;
+		uintptr_t retValue = 0;
+
+		if (oldValue) {
+			return false;
+		}
+
+		/* atomically low tag ('set', not 'or') continuation->carrierThread */
+		if ((retValue = VM_AtomicSupport::lockCompareExchange(localAddr, oldValue, J9_GC_CONTINUATION_CONCURRENTSCANNING)) != oldValue) {
+			if (J9_ARE_ANY_BITS_SET(retValue, ~(uintptr_t)J9_GC_CONTINUATION_CONCURRENTSCANNING)) {
+				/* low tagging failed due to mounting, we don't need to do anything - it will be compensated by pre/post mount actions, depending on GC style */
+
+			} else {
+				/* low tagging failed due to another GC thread winning low tagging, we don't do anything - winning thread will do everything instead */
+
+			}
```

Comment:
I'd structure the code like this:

```
if (0 == oldValue) {
	if (atomic succeeds) 
            return true;
}

return false;
```

And I'd put just one comment in front of the method (since it applies to both not attempting atomic and attempting, but losing) like:

If low tagging failed (or not even attempted) due to either 
- a carrier thread winning to mount, we don't need to do anything, since it will be compensated by pre/post mount actions
- another GC thread winning to scan, again don't do anything, and let the winning thread do the work, instead